apps/studio/pages/new/[slug].tsx

```diff
@@ -779,41 +778,58 @@ const Wizard: NextPageWithLayout = () => {
-                                    await form.trigger('dbPass')
-                                  } else {
-                                    await delayedCheckPasswordStrength(value)
-                                  }
-                                }}
-                              />
-                            </FormControl_Shadcn_>
-                          </FormItemLayout>
-                        )}
+                        render={({ field }) => {
+                          const regex = /^[^@:\/]*$/
+                          const hasSpecialCharacters =
+                            field.value.length > 0 && !field.value.match(regex)
+
+                          return (
+                            <FormItemLayout
+                              label="Database Password"
+                              layout="horizontal"
+                              description={
+                                <>
+                                  {hasSpecialCharacters && (
+                                    <p className="mb-2">
+                                      Note: Special symbols need to be{' '}
+                                      <InlineLink href="https://supabase.com/docs/guides/database/postgres/roles#special-symbols-in-passwords">
+                                        percent-encoded
+                                      </InlineLink>{' '}
+                                      when using your password later in Postgres connection strings
+                                    </p>
+                                  )}
+                                  <PasswordStrengthBar
+                                    passwordStrengthScore={form.getValues('dbPassStrength')}
+                                    password={field.value}
+                                    passwordStrengthMessage={passwordStrengthMessage}
+                                    generateStrongPassword={generatePassword}
+                                  />
+                                </>
+                              }
+                            >
+                              <FormControl_Shadcn_>
+                                <Input
+                                  copy={field.value.length > 0}
+                                  type="password"
+                                  placeholder="Type in a strong password"
+                                  {...field}
+                                  autoComplete="off"
+                                  onChange={async (event) => {
+                                    field.onChange(event)
+                                    form.trigger('dbPassStrength')
+                                    const value = event.target.value
+                                    if (event.target.value === '') {
+                                      await form.setValue('dbPassStrength', 0)
+                                      await form.trigger('dbPass')
```

Comment:
i know this wasn't in the PR, but this is strange.
seems like we should just form.trigger('dbPassStrength') regardless, and not check for a value to be defined for this input

Reply:
this is to immediately set the password strength bar to 0 if the password value is empty
otherwise the checking of password strength is on a debounce